### DISCUSSION TITLE
Composited Label in Windows BAT file

### DISCUSSION THREAD

**Discussion by johnb2233** [ImageMagick/ImageMagick#2131] [Help] [answered]
I found this example of a Composited Label:

    convert -background '#00000080' -fill white label:'Faerie Dragon' miff:- |\
        composite -gravity south -geometry +0+3 \
                  -   dragon.gif   anno_composite.jpg

Its exactly what I want to do in batch file, except for a black background and source fiule being JPG

I've tried converting it to a bat command with no luck. Can someone please convert this to bat command.

Magick version is 7.0.8.42

Many thanks in advance

John

**Comment by GeeMack** [comment]
First, since you're using ImageMagick version 7, you should use "magick" instead of "convert".  Also, small syntax difference between IMv6 and IMv7 might make some of the online instructions for v6 a little confusing when you're working with v7.

ImageMagick commands are fairly simple to translate from *nix to Windows. Usually you can change the continued line backslashes \ to carets ^, change (most) single quotes ' to double quotes ", and if it's going in a Windows BAT script, the percent signs % need to be (mostly) doubled %%.

For your command I would run it as a single "magick" command, and not bother with the pipe. Here is a Windows IM command that does the same thing as your example.

    magick dragon.gif ^
       -background "#00000080" -fill white label:"Faerie Dragon" ^
       -gravity south -geometry +0+3 -composite anno_composite.jpg

That reads in the "dragon.gif" image, creates the label, sets the gravity and geometry, and composites the label at the bottom of the input image. Pretty straightforward.

**Comment by johnb2233** [comment, marked as answer]
GeeMack, 
Many thanks great solution.  As we say downunder your bloods worth bottling.

Have a great weekend and stay safe.
